MVP_Clicker/Assets/Project/Scripts/Game/Areas/BonusesShop/Model/BonusesShopModel.cs

```diff
@@ -48,26 +49,43 @@ private void UpdateBonusInfo(string upgradedBonus)
 
         private void PayForUpgrade(string upgradedBonus)
         {
-            _gameResources.Collection[_gameResourcesId.Money].Amount -= _collection[upgradedBonus].UpgradeValue;
+            _gameResources.Collection[_gameResourcesConfig.Money.Id].Amount -= _collection[upgradedBonus].UpgradeValue;
         }
 
         private void UpdateDamagePerTapBonus()
         {
-            int sumDamagePerTapBonusFromAllBonuses = _collection[_bonusId.Sword].ProvidingDamagePerTapBonus;
+            int sumDamagePerTapBonusFromAllBonuses = CalculateDamagePerTapBonusFromAllBonuses(_collection);
             int receivedDamagePerTapBonus = sumDamagePerTapBonusFromAllBonuses;
-            _gameResources.Collection[_gameResourcesId.DamagePerTap].Amount = receivedDamagePerTapBonus;
+            _gameResources.Collection[_gameResourcesConfig.DamagePerTap.Id].Amount = receivedDamagePerTapBonus;
         }
 
         private void AddListeners()
         {
-            _collection[_bonusId.Sword].UpgradeBought += OnUpgradeBought;
-            _collection[_bonusId.Sword].DamagePerTapBonusChanged += UpdateDamagePerTapBonus;
+            foreach (KeyValuePair<string, IBonusModel> bonus in _collection)
```

Comment:
You can do look more good if you will change it to
`foreach (var bonus in _collection)`

But it's not critical